Arm: deprecated warnings happen on samples

##### System information (version)
- OpenCV => recent master, 3.4 ( 1894f1a37 )
- Operating System / Platform => Debian 9.4 32bit / TinkerBoard
- Compiler => GCC 6.3.0

##### Detailed description

The build log contains warning about deprecated API.

```
[ 47%] Building CXX object samples/opencl/CMakeFiles/example_opencl_opencl-opencv-interop.dir/opencl-opencv-interop.cpp.o
/home/linaro/opencv-fork/samples/opencl/opencl-opencv-interop.cpp: In member function ‘int App::initOpenCL()’:
/home/linaro/opencv-fork/samples/opencl/opencl-opencv-interop.cpp:594:19: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
         m_queue = clCreateCommandQueue(m_context, m_device_id, 0, &res);
                   ^~~~~~~~~~~~~~~~~~~~
In file included from /home/linaro/opencv-fork/samples/opencl/opencl-opencv-interop.cpp:22:0:
/usr/include/CL/cl.h:1427:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
```

https://github.com/opencv/opencv/blob/1894f1a37f8bdf06788cb04e0ab538a9be970f17/samples/opencl/opencl-opencv-interop.cpp#L594

Now, this doesn't happen in the OpenCV it self, only during the build of sample program.

What's the difference ?
The difference was [here](https://github.com/opencv/opencv/blob/1894f1a37f8bdf06788cb04e0ab538a9be970f17/samples/opencl/opencl-opencv-interop.cpp#L17)

https://github.com/opencv/opencv/blob/1894f1a37f8bdf06788cb04e0ab538a9be970f17/samples/opencl/opencl-opencv-interop.cpp#L17

This ```CL_USE_DEPRECATED_OPENCL_2_0_APIS``` macro enables the warning of old APIs.

Now, Tinkerboard only supports OpenCL up to 1.2, not 2.0.
Here's a log from ```opencv_test_core```
```
OpenCL Platforms: 
    ARM Platform
        iGPU: Mali-T760 (OpenCL 1.2 v1.r9p0-05rel0-git(f980191).e4ba9e4c6ff8005348d0332aae160089)
```

I also looked when this define was introduced, and it was #5241.
Unfortunately, there were no description about the PR.  I'm not sure which build warning was supposed to be eliminated.


Just to make sure, I think we can modify this macro to ```CL_USE_DEPRECATED_OPENCL_1_2_APIS``` since Arm GPU hardly supports 2.0 at this stage. ([cf](https://stackoverflow.com/questions/28500496/opencl-function-found-deprecated-by-visual-studio))

Or, we may remove this for lower level OpenCL.

Which way should I go ?


##### Steps to reproduce

```
mkdir build
cd build
time cmake -DBUILD_EXAMPLES:BOOL=ON -DCPU_BASELINE=FP16 -DNEON=ON .. 
time make -j 4 all
```

<!-- to add code example fence it with triple backticks and optional file extension
    ```.cpp
    // C++ code example
    ```
 or attach as .txt or .zip file
-->

--- a/samples/opencl/opencl-opencv-interop.cpp
+++ b/samples/opencl/opencl-opencv-interop.cpp
@@ -14,6 +14,8 @@
 #include <iomanip>
 #include <stdexcept>
 
+#define CL_USE_DEPRECATED_OPENCL_1_1_APIS
+#define CL_USE_DEPRECATED_OPENCL_1_2_APIS
 #define CL_USE_DEPRECATED_OPENCL_2_0_APIS // eliminate build warning
 
 #ifdef __APPLE__

